### ISSUE TITLE
TCP instrumentation: Cannot read property 'apply' of undefined

### ISSUE THREAD

**Issue by pukhalski** [opbeat/opbeat-node#64]
```
TypeError: Cannot read property 'apply' of undefined
    at Command.wrappedCallback (/node_modules/opbeat/lib/instrumentation/modules/redis.js:36:20)
    at Command.instrumented [as callback] (/node_modules/opbeat/lib/instrumentation/index.js:112:27)
    at normal_reply (/node_modules/redis/index.js:714:21)
    at RedisClient.return_reply (/node_modules/redis/index.js:816:9)
    at JavascriptRedisParser.Parser.returnReply (/node_modules/redis/index.js:188:18)
    at JavascriptRedisParser.execute (/node_modules/redis-parser/lib/parser.js:415:12)
    at Socket.<anonymous> (/node_modules/redis/index.js:267:27)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:177:18)
    at Socket.Readable.push (_stream_readable.js:135:10)
    at TCP.onread (net.js:542:20)
    at TCP.instrumented [as onread] (/node_modules/opbeat/lib/instrumentation/index.js:112:27)
```

**Comment by roncohen** [comment]
Hi @pukhalski. Which version of the `redis` module are you using?

**Comment by pukhalski** [comment]
Hey @roncohen, thanks for the fast reply.

```"redis": "^2.6.0-2"``` by `https://github.com/wyattjoh/rate-limit-redis/`

**Comment by pukhalski** [comment]
Confirmed. This library causes the issue.

**Comment by joeycozza** [comment]
Also having the issue, using redis 2.6.2. 

Here is the offending code.
```javascript
if (trace && commandObj) {
        var cb = commandObj.callback
        commandObj.callback = opbeat._instrumentation.bindFunction(function wrappedCallback () {
          trace.end()
          return cb.apply(this, arguments)
        })
        trace.start(String(command).toUpperCase(), 'cache.redis')
      }
```

I forked and just added to the if statement ```&& commandObj.callback``` for now. 

I don't know if you'd care for a PR with that tiny change, or if you'd rather find/get to the root of the problem and make the instrumentation actually work.

Let me know if I can be of any assistance. Would love to help out if I can.

**Comment by roncohen** [comment]
@joeycozza Thanks for the investigation. It looks like the optional callback is the cause of the problem. Even if the user code does not set a callback, it could be useful to time these commands instead of just getting rid of them. When the callback does not exist, we could set it and only do `trace.end()` in there.

**Comment by roncohen** [comment]
@pukhalski @joeycozza hey guys. Could you try out https://github.com/opbeat/opbeat-node/pull/65 and see if it fixes your issue?

**Comment by roncohen** [comment]
Fixed in 3.14.1. Let me know if you see any more issues.
